mozilla/addons-server
# rm AbuseReport.appeallant_job

Fixes: mozilla/addons#14984

~blocked on https://github.com/mozilla/addons-server/pull/22582 - there will be merge conflicts.~



### Description

Drops the `appellant_job` fk, as well as `reporter_appeal_date`, that was effectively unused (it was referenced in one place where `appellant_job` could have been used, and now `cinderappeal` can be instead).  DB migration to drop columns from the table can come in a later deploy.

### Testing

No particular testing, as it should be functionally the same, but to confirm you can:
- test the migration by:
  - have an abuse report with an original job and a decision that was appealed (so a new, appeal, cinder job), prior to checking out this patch (the AbuseReport would have `appellant_job` set to the appeal cinder job)
  - check out the patch so the migration runs
  - see there is now a CinderAppeal instance
- test abuse-related functionality by:
  - appealing a decision on a job as the reporter
  - checking you can't appeal the decision twice as a reporter
  - seeing that emails are sent out to reporter(s) that appeal a decision

### Checklist
